test/units/executor/test_action_write_locks.py

```diff
@@ -0,0 +1,99 @@
+# (c) 2021, Kevin Carter <kecarter@redhat.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import sys
+
+from multiprocessing import Process
+
+from multiprocessing.managers import DictProxy
+
+import pytest
+
+from ansible.executor import action_write_locks
+from ansible.module_utils import six
+
+
+ACTION_LOCK = action_write_locks.action_lock
+ACTION_WRITE_LOCKS = action_write_locks.action_write_locks
+
+
+def _run_threads(test_int):
+    """Run thread test.
+
+    Test that that the original value is in our shared
+    dictionary and that we can add items into the cache
+    from within a process.
+    """
+    assert "testX" in ACTION_WRITE_LOCKS
+    ACTION_WRITE_LOCKS["test%i" % test_int] = ACTION_LOCK()
+
+
+def test_write_lock():
+    """Test that the write lock is created with default items."""
+    default_locks = [
+        "copy",
+        "file",
+        "setup",
+        "slurp",
+        "stat",
+    ]
+    for item in default_locks:
+        assert item in ACTION_WRITE_LOCKS
+
+    if sys.version_info > (3, 0, 0):
+        assert type(ACTION_WRITE_LOCKS) == DictProxy
+    else:
+        assert type(ACTION_WRITE_LOCKS) == dict
+
+
+def test_new_lock():
+    """Test a new lock can be created."""
+    ACTION_WRITE_LOCKS["test"] = ACTION_LOCK()
+    assert "test" in ACTION_WRITE_LOCKS
+
+
+@pytest.mark.skipif(six.PY2, reason="Test requires python 3")
+def test_new_lock_with_threads_and_reimport():
+    """Test that a new lock can be created within a Process.
+
+    Note: this test is only ever fully executed when running PY3.
+    """
+    ACTION_WRITE_LOCKS["testX"] = ACTION_LOCK()
+
+    active_processes = list()
+    processes = [
+        Process(target=_run_threads, args=(1,)),
+        Process(target=_run_threads, args=(2,)),
+        Process(target=_run_threads, args=(3,)),
+        Process(target=_run_threads, args=(4,)),
+        Process(target=_run_threads, args=(5,)),
+    ]
+    for t in processes:
+        t.daemon = True
+        active_processes.append(t)
+        t.start()
+
+    for t in active_processes:
+        t.join()
```

Comment:
I don't think another var is necessary. If the first iteration to start the processes succeeds and won't interrupt the control flow, all of them will be active.
```suggestion
    processes = [
        Process(target=_run_threads, args=(_,))
        for _ in range(1, 6)
    ]
    for t in processes:
        t.daemon = True
        t.start()

    for t in processes:
        t.join()
```